api/src/test/java/org/pmiops/workbench/api/WorkspacesControllerTest.java

```diff
@@ -536,6 +536,44 @@ public void testUpdateWorkspace() throws Exception {
     assertThat(got).isEqualTo(ws);
   }
 
+  @Test
+  public void testUpdateWorkspaceResearchPurpose() throws Exception {
+    Workspace ws = createDefaultWorkspace();
+    ws = workspacesController.createWorkspace(ws).getBody();
+
+    ResearchPurpose rp = new ResearchPurpose()
+            .diseaseFocusedResearch(false)
+            .diseaseOfFocus(null)
+            .methodsDevelopment(false)
+            .controlSet(false)
+            .aggregateAnalysis(false)
+            .ancestry(false)
+            .commercialPurpose(false)
+            .population(false)
+            .populationOfFocus(null)
+            .additionalNotes(null)
+            .reviewRequested(false);
+    ws.setResearchPurpose(rp);
+    UpdateWorkspaceRequest request = new UpdateWorkspaceRequest();
+    request.setWorkspace(ws);
+    ResearchPurpose updatedRp = workspacesController
+            .updateWorkspace(ws.getNamespace(), ws.getId(), request)
+            .getBody()
+            .getResearchPurpose();
+
+    assertThat(updatedRp.getDiseaseFocusedResearch()).isFalse();
```

Comment:
if most of these things default to false (i assume) should we try this test setting some of them to true?

Reply:
They are all set to true/truthy in the create default workspace method, hence my setting them to false here.

Reply:
You know what I can do though ... is to reset them again in a mixed fashion and ensure they still line up.

Reply:
ah that makes sense.  no need to change then unless you want to!

Reply:
Ok, thanks ... was having issues making multiple update calls inside the same test. Will go with this and we can always add more tests.